matchms/importing/load_from_msp.py

```diff
@@ -149,16 +147,20 @@ def parse_metadata(rline: str, params: dict):
     matches = []
     splitted_line = rline.split(":", 1)
     if splitted_line[0].lower() == 'comments' and "=" in splitted_line[1]:
-        pattern = r'(\S+)="([^"]*)"|"(\w+)=([^"]*)"|"([^"]*)=([^"]*)"|(\S+)=(\d+(?:\.\d*)?)'
+        pattern = r'(\S+)="([^"]+)"|' \
+                  r'"(\w+)=([^"]+)"|' \
+                  r'"([^"]+)=([^"]+)"|' \
+                  r'(\S+)=(\d+(?:\.\d*)?)'
         matches = re.findall(pattern, splitted_line[1].replace("'", '"'))
         for match in matches:
             match = [i for i in match if i]
```

Comment:
This regex will return a tuple with 8 matches, of which 6 will be empty ''. Only the non empty once are stored. This was what made the code break at first, since cases like "compound_name=" Would have an empty value for 7 tuples, resulting in match[1] not existing. The reason this regex is so complex, is that it tries to cover different formats, to be specific different usages of "". It can be: "compound_name=caffeine" compound_name="caffeine", and parent_mass=12.3 I found this structure a bit hacky, but did not really know a quick fix, so just added the check that len(match) ==2

Reply:
I think the pattern "(\w+)=([^"]+)" is always captured by "([^"]+)=([^"]+)" as well. So I did remove this, since this does not add anything.